lnwallet/rpcwallet/remote_signer_client.go

```diff
@@ -0,0 +1,953 @@
+			return nil, err
+		}
+
+		rType := &walletrpc.SignCoordinatorResponse_MuSig2CombineSigResponse{
+			MuSig2CombineSigResponse: resp,
+		}
+
+		signResp.SignResponseType = rType
+
+		return signResp, nil
+
+	case *walletrpc.SignCoordinatorRequest_MuSig2CleanupRequest:
+		resp, err := r.signerServer.MuSig2Cleanup(
+			ctx, reqType.MuSig2CleanupRequest,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		rType := &walletrpc.SignCoordinatorResponse_MuSig2CleanupResponse{
+			MuSig2CleanupResponse: resp,
+		}
+
+		signResp.SignResponseType = rType
+
+		return signResp, nil
+
+	case *walletrpc.SignCoordinatorRequest_SignPsbtRequest:
+		resp, err := r.walletServer.SignPsbt(
+			ctx, reqType.SignPsbtRequest,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		rType := &walletrpc.SignCoordinatorResponse_SignPsbtResponse{
+			SignPsbtResponse: resp,
+		}
+
+		signResp.SignResponseType = rType
+
+		return signResp, nil
+
+	case *walletrpc.SignCoordinatorRequest_Ping:
+		// If the received request is a ping, we don't need to pass the
+		// request on to a server, but can respond with a pong directly.
+		rType := &walletrpc.SignCoordinatorResponse_Pong{
+			Pong: true,
+		}
+
+		signResp.SignResponseType = rType
+
+		return signResp, nil
+
+	default:
+		return nil, ErrRequestType
+	}
+}
+
+// sendResponse sends the passed response back to the watch-only node over the
+// stream.
+func (r *OutboundClient) sendResponse(ctx context.Context, resp *signerResponse,
+	stream *Stream) error {
+
+	// We send the response in a goroutine to ensure we can return an error
+	// if the send times out or the context is canceled. This is done to
+	// ensure that this function won't block indefinitely.
+	var (
+		sendDone = make(chan struct{})
+		errChan  = make(chan error)
+
+		// The returnedChan is used to signal that this function has
+		// already returned, so that the goroutines don't remain blocked
+		// indefinitely when trying to send to the above channels.
+		returnedChan = make(chan struct{})
+	)
+	defer close(returnedChan)
+
+	success := r.gManager.Go(func(_ context.Context) {
+		err := stream.Send(resp)
+		if err != nil {
+			select {
+			case errChan <- err:
+			case <-returnedChan:
+			}
+
+			return
+		}
+
+		close(sendDone)
+	})
+	if !success {
+		return fmt.Errorf("error starting send function")
+	}
+
+	select {
+	case err := <-errChan:
+		return fmt.Errorf("send response to watch-only node error: %w",
+			err)
+
+	case <-time.After(r.requestTimeout):
+		return errors.New("send response to watch-only node timeout")
+
+	case <-r.quit:
+		return ErrShuttingDown
+
+	case <-ctx.Done():
+		return ctx.Err()
+
+	case <-sendDone:
+		return nil
+	}
+}
+
+// A compile time assertion to ensure OutboundClient meets the
+// RemoteSignerClient interface.
+var _ RemoteSignerClient = (*OutboundClient)(nil)
+
+// formatSignCoordinatorMsg formats the passed proto message into a JSON string.
+func formatSignCoordinatorMsg(msg protoreflect.ProtoMessage) string {
```

Comment:
note that this builds the string regardless of log level. Should use a log closure instead

Reply:
see patch